<?xml version="1.0" encoding="utf-8"?>
<launch>

    <arg name="controller_name" default="iiwa_position_controller" />

    <arg name="position_command_topic" default="joint_command_position" />
    <arg name="status_topic" default="$(arg controller_name)/status" />
    <arg name="config_feedback_topic" default="joint_states" />
    <arg name="velocity_command_topic" default="joint_command_velocity" />
    <arg name="abort_service" default="$(arg controller_name)/abort" />

    <arg name="control_rate" default="100.0" />
    <arg name="velocity_limit_scaling" default="0.75" />

    <arg name="output" default="screen" />
    <arg name="use_sim_time" default="false" />
    <arg name="remapped_clock_topic" default="/clock" />

    <!-- Start the position controller -->
    <node pkg="iiwa_robot_controllers" type="$(arg controller_name)" name="$(arg controller_name)" output="$(arg output)">
        <param name="position_command_topic" type="string" value="$(arg position_command_topic)" />
        <param name="status_topic" type="string" value="$(arg status_topic)" />
        <param name="config_feedback_topic" type="string" value="$(arg config_feedback_topic)" />
        <param name="velocity_command_topic" type="string" value="$(arg velocity_command_topic)" />
        <param name="abort_service" type="string" value="$(arg abort_service)" />
        <param name="control_rate" type="double" value="$(arg control_rate)" />
        <param name="velocity_limit_scaling" type="double" value="$(arg velocity_limit_scaling)" />
        <param name="use_sim_time" type="bool" value="$(arg use_sim_time)" />
        <remap from="/clock" to="$(arg remapped_clock_topic)"/>
    </node>

</launch>
